### PR TITLE
Modifies notification for MUC conversations

### PR DESCRIPTION
**Pull request by badarsh2**
Fix for issue https://github.com/siacs/Conversations/issues/2342
Adds the message senders in MUC notifications

Before:
fd4a48fe-018b-11e7-9c7a-3449dc88ce28

After:
screenshot_20170306-080408

### CODE REVIEW

File: src/main/java/eu/siacs/conversations/services/NotificationService.java [hunk 1]
```diff
@@ -388,8 +388,25 @@ private void modifyForTextOnly(final Builder builder, final ArrayList<Message> m
...
+				for (Message message : messages) {
+					final String name = message.getContact().getDisplayName();
+					style.addLine(Html.fromHtml("<b>"+name+"</b>: "+message.getBody()));
+				}
+				builder.setStyle(style);
+				if(messages.size() == 1) {
+					builder.setContentText(messages.get(0).getBody());
+				}
+				else {
+					builder.setContentText(messages.size() + " unread messages");
```
Comment (iNPUTmice) [inline src/main/java/eu/siacs/conversations/services/NotificationService.java]: Please translate this string. There is already a translatable string for this.

File: src/main/java/eu/siacs/conversations/services/NotificationService.java [hunk 2]
```diff
@@ -388,8 +388,25 @@ private void modifyForTextOnly(final Builder builder, final ArrayList<Message> m
...
-			builder.setContentText(UIHelper.getMessagePreview(mXmppConnectionService, messages.get((messages.size()-1))).first);
+			if(messages.get(0).getConversation().getMode() == Conversation.MODE_SINGLE) {
+				Log.d("NotificationService", messages.get(0).getContact().getDisplayName());
+				builder.setStyle(new NotificationCompat.BigTextStyle().bigText(getMergedBodies(messages)));
+				builder.setContentText(UIHelper.getMessagePreview(mXmppConnectionService, messages.get((messages.size() - 1))).first);
+			}
+			else {
+				final NotificationCompat.InboxStyle style = new NotificationCompat.InboxStyle();
+				for (Message message : messages) {
+					final String name = message.getContact().getDisplayName();
```
Comment (iNPUTmice) [inline src/main/java/eu/siacs/conversations/services/NotificationService.java]: Message.getContact() will be null for public MUC.
There is a helper method `getMessageDisplayName(final Message message)` that automatically does the right thing.

File: src/main/java/eu/siacs/conversations/services/NotificationService.java [hunk 3]
```diff
@@ -388,8 +388,25 @@ private void modifyForTextOnly(final Builder builder, final ArrayList<Message> m
 			}
 			builder.setStyle(messagingStyle);
 		} else {
-			builder.setStyle(new NotificationCompat.BigTextStyle().bigText(getMergedBodies(messages)));
-			builder.setContentText(UIHelper.getMessagePreview(mXmppConnectionService, messages.get((messages.size()-1))).first);
+			if(messages.get(0).getConversation().getMode() == Conversation.MODE_SINGLE) {
+				Log.d("NotificationService", messages.get(0).getContact().getDisplayName());
```
Comment (iNPUTmice) [inline src/main/java/eu/siacs/conversations/services/NotificationService.java]: Please remove that logging output. It's fine when you need to to figure out what's going on but it serves no purpose in production.

**Comment by badarsh2** [comment]
@iNPUTmice I have made the changes as requested.

**Comment by iNPUTmice** [comment]
Actually I just realized that; Can you use a SpannableString please for the formatting instead of HTML.fromHtml()? I realize I'm doing the same in my code as well; but that's also bad. And while you are at it you could also fix the other code as well? (It's virtually the same code so that should be more effort)

**Comment by badarsh2** [comment]
@iNPUTmice Done!

**Pull request merged**